reframe/frontend/statistics.py

```diff
@@ -1,89 +1,103 @@
 import reframe.core.debug as debug
+from reframe.core.exceptions import (ReframeFatalError, StatisticsError)
 
 
 class TestStats:
     """Stores test case statistics."""
 
-    def __init__(self, tasks=[]):
-        if not hasattr(tasks, '__iter__'):
-            raise TypeError('expected an iterable')
-
-        # Store test cases per partition internally
-        self._tasks_bypart = {}
-        for t in tasks:
-            partition = t.check.current_partition
-            partname = partition.fullname if partition else 'None'
-            tclist = self._tasks_bypart.setdefault(partname, [])
-            tclist.append(t)
+    def __init__(self):
+        self._tasks = [[]]
+        self._current_run = 0
 
     def __repr__(self):
         return debug.repr(self)
 
-    def num_failures(self, partition=None):
-        num_fails = 0
-        if partition:
-            num_fails += len([
-                t for t in self._tasks_bypart[partition] if t.failed
-            ])
-        else:
-            # count all failures
-            for tclist in self._tasks_bypart.values():
-                num_fails += len([t for t in tclist if t.failed])
-
-        return num_fails
-
-    def num_failures_stage(self, stage):
-        num_fails = 0
-        for tclist in self._tasks_bypart.values():
-            num_fails += len([t for t in tclist if t.failed_stage == stage])
-
-        return num_fails
-
-    def num_cases(self, partition=None):
-        num_cases = 0
-        if partition:
-            num_cases += len(self._tasks_bypart[partition])
-        else:
-            # count all failures
-            for tclist in self._tasks_bypart.values():
-                num_cases += len(tclist)
-
-        return num_cases
+    def next_run(self, current_run):
```

Comment:
As I said before, this should not take an argument and the check below should be removed.

Reply:
You can instead add a property that reads out the `current_run`.

Reply:
Done.